frontend/packages/integration-tests-cypress/tests/app/demo-dynamic-plugin.spec.ts

```diff
@@ -0,0 +1,198 @@
+  cy.get('.pf-c-title.pf-m-2xl').should('contain', `Dynamic Page ${navID}`);
+  cy.get(infoMessage).should('contain', 'Example info alert');
+  cy.get(warningMessage).should('contain', 'Example warning alert');
+  cy.get(hintMessage).should('contain', 'Example hint');
+  cy.get('.pf-c-card').should('contain', 'Example card');
+};
+
+const k8sAPINavTest = (apiID: string) => {
+  cy.byButtonText(apiID).click();
+  cy.get('test-k8api-error').should('not.exist');
+  cy.get(`test-k8s-${apiID}`).should('not.be.empty');
+};
+
+describe('Demo dynamic plugin test', () => {
+  before(() => {
+    cy.login();
+    nav.sidenav.switcher.changePerspectiveTo('Administrator');
+    nav.sidenav.switcher.shouldHaveText('Administrator');
+    cy.createProject(PLUGIN_NAME);
+    cy.readFile(`${PLUGIN_PATH}/oc-manifest.yaml`).then((textManifest) => {
+      const yamlManifest = safeLoadAll(textManifest);
+      const deployment = yamlManifest.find(({ kind }) => kind === 'Deployment');
+
+      if (!isLocalDevEnvironment && PLUGIN_PULL_SPEC) {
+        console.log('this is not a local env, setting the pull spec for the deployment');
+        deployment.spec.template.spec.containers[0].image = PLUGIN_PULL_SPEC;
+      } else {
+        console.log('this IS A local env, not setting the pull spec for the deployment');
+      }
+      const service = yamlManifest.find(({ kind }) => kind === 'Service');
+      const consolePlugin = yamlManifest.find(({ kind }) => kind === 'ConsolePlugin');
+      cy.exec(` echo '${JSON.stringify(deployment)}' | oc create -f -`, {
+        failOnNonZeroExit: false,
+      })
+        .its('stdout')
+        .should('contain', 'created')
+        .then(() =>
+          cy
+            .exec(` echo '${JSON.stringify(service)}' | oc create -f -`, {
+              failOnNonZeroExit: false,
+            })
+            .then((result) => {
+              console.log('Error: ', result.stderr);
+              console.log('Success: ', result.stdout);
+            })
+            .its('stdout')
+            .should('contain', 'created'),
+        )
+        .then(() =>
+          cy
+            .exec(` echo '${JSON.stringify(consolePlugin)}' | oc create -f -`, {
+              failOnNonZeroExit: false,
+            })
+            .then((result) => {
+              console.log('Error: ', result.stderr);
+              console.log('Success: ', result.stdout);
+            })
+            .its('stdout')
+            .should('contain', 'created'),
+        )
+        .then(() => {
+          cy.visit(`/k8s/ns/${PLUGIN_NAME}/deployments`);
+          listPage.rows.shouldBeLoaded();
+          listPage.filter.byName(PLUGIN_NAME);
+          listPage.rows.shouldExist(PLUGIN_NAME);
+          enableDemoPlugin(true);
+        });
+    });
+  });
+
+  afterEach(() => {
+    checkErrors();
+  });
+
+  after(() => {
+    enableDemoPlugin(false);
+    cy.deleteProject(PLUGIN_NAME);
+    cy.logout();
```

Comment:
We also need remove consoleplugin/console-demo-plugin

Reply:
@yapei yes thats correct. adding / removing the plugin is on line 76 of the enable/disable plugin function that is called on line 160.